src/runtime/security/server_negotiation.cpp

```diff
@@ -112,12 +112,5 @@ error_s server_negotiation::do_sasl_server_init()
     return error_s::make(ERR_OK);
 }
 
-void server_negotiation::fail_negotiation(negotiation_rpc rpc, const std::string &reason)
-{
-    negotiation_response &response = rpc.response();
-    _status = response.status = negotiation_status::type::SASL_AUTH_FAIL;
-    response.msg = reason;
```

Comment:
In previous implementation, there will be error reason. Is it totally useless?

Reply:
It is only used for sending a message to client to show why it is failed. I think it is useless